.github/workflows/code_test.yaml

```diff
@@ -0,0 +1,70 @@
+---
+name: Code test
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  integration-tests:
+    name: Run integration tests
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out repository
+        uses: actions/checkout@v4
+
+      - name: Install rust toolchain
+        uses: actions-rust-lang/setup-rust-toolchain@v1
+
+      - name: Run integration tests
+        run: cargo test --tests
+
+      - name: Install cargo-tarpaulin
+        run: cargo install cargo-tarpaulin
+
+      - name: Run unit tests with coverage
+        run: cargo tarpaulin --tests --out Xml
```

Comment:
I think you should also add the `--doc` flag so that tarpaulin includes codeblocks from the docstrings